sunpy/net/jsoc/tests/test_jsoc.py

```diff
@@ -318,3 +318,16 @@ def test_results_filenames():
     assert len(files) == len(responses)
     for hmiurl in files:
         assert os.path.isfile(hmiurl)
+
+
+@pytest.mark.remote_data
+def test_results_filenames_as_is(tmp_path):
+    responses = client.search(
+        a.Time('2014/1/1T1:00:36', '2014/1/1T01:01:38'),
+        a.jsoc.Series('hmi.M_45s'), a.jsoc.Notify('jsoc@cadair.com'),
+        a.jsoc.Protocol('as-is'))
+    files = client.fetch(responses, path=tmp_path)
```

Comment:
Can you add an:
```python
assert len(responses) == X
```
here to check we are getting the correct number of results before we download them?